pkg/pipeline/templates/controller.go.tmpl

```diff
@@ -44,7 +40,7 @@ func Setup(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter, s terra
 
 	return ctrl.NewControllerManagedBy(mgr).
 		Named(name).
-		WithOptions(controller.Options{RateLimiter: rl, MaxConcurrentReconciles: concurrency}).
+		WithOptions(o.ForControllerRuntime()).
 		For(&{{ .TypePackageAlias }}{{ .CRD.Kind }}{}).
 		Complete(r)
```

Comment:
```suggestion
		Complete(ratelimiter.NewReconciler(name, r, o.GlobalRateLimiter))
```

This is needed to ensure the controllers are "globally" rate limited - i.e. that even non-error / `Requeue: true` cases are subject to rate limiting.

Reply:
Done, thank you for catching!